packages/web-api-scan-request-sender/src/sender/dispatcher.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { client, CosmosOperationResponse } from 'azure-services';
+import { GuidGenerator, HashGenerator, ServiceConfiguration } from 'common';
+import { inject, injectable } from 'inversify';
+import { Logger } from 'logger';
+import { PageScanRequestProvider } from 'service-library';
+import { OnDemandPageScanRequest } from 'storage-documents';
+import { ScanRequestSender } from './scan-request-sender';
+
+@injectable()
+export class DDispatcher {
+    constructor(
+        @inject(PageScanRequestProvider) private readonly pageScanRequestProvider: PageScanRequestProvider,
+        @inject(Logger) private readonly logger: Logger,
+        @inject(ScanRequestSender) private readonly sender: ScanRequestSender,
+        @inject(ServiceConfiguration) private readonly serviceConfig: ServiceConfiguration,
+        @inject(HashGenerator) private readonly hashGenerator: HashGenerator,
+        @inject(GuidGenerator) private readonly guidGenerator: GuidGenerator,
+    ) {}
+
+    public async dispatchOnDemandScanRequests(): Promise<void> {
+        const configQueueSize = (await this.serviceConfig.getConfigValue('queueConfig')).maxQueueSize;
+        this.logger.logInfo(`[Sender] Maximum queue size configuration set to ${configQueueSize}`);
+        console.log(`[Sender] Maximum queue size configuration set to ${configQueueSize}`);
+        let currentQueueSize = await this.sender.getCurrentQueueSize();
+        this.logger.logInfo(`[Sender] Current queue size is ${currentQueueSize}`);
+        if (currentQueueSize >= configQueueSize) {
+            this.logger.logWarn('[Sender] Unable to queue new scan request as queue already reached to its maximum capacity');
```

Comment:
>logWarn [](start = 24, length = 7)

this shouldnt be warning. change it to info

Reply:
Discussed offline. Keeping as is to aid debugging. This feels like something that should stand out when debugging. Easily ignorable when debugging.